sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py

```diff
@@ -35,22 +47,26 @@ def __init__(self, topic_hostname, credential, **kwargs):
 
 
     async def send(self, events, **kwargs):
-        # type: (Union[List[CloudEvent], List[EventGridEvent], List[CustomEvent]], Any) -> None
+        # type: (SendType) -> None
         """Sends event data to topic hostname specified during client initialization.
 
         :param  events: A list of CloudEvent/EventGridEvent/CustomEvent to be sent.
         :type events: Union[List[models.CloudEvent], List[models.EventGridEvent], List[models.CustomEvent]]
         :rtype: None
         raise: :class:`ValueError`, when events do not follow specified SendType.
          """
+        if not isinstance(events, list):
+            events = [events]
 
-        if all(isinstance(e, CloudEvent) for e in events):
-            await self._client.publish_cloud_event_events(self._topic_hostname, events, **kwargs)
-        elif all(isinstance(e, EventGridEvent) for e in events):
-            await self._client.publish_events(self._topic_hostname, events, **kwargs)
+        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+            kwargs.setdefault("content_type", "application/cloudevents-batch+json; charset=utf-8")
+            self._client.publish_cloud_event_events(self._topic_hostname, events, **kwargs)
+        elif all(isinstance(e, EventGridEvent) for e in events) or all(isinstance(e, dict) for e in events):
+            kwargs.setdefault("content_type", "application/json; charset=utf-8")
+            self._client.publish_events(self._topic_hostname, events, **kwargs)
```

Comment:
Nit: seeing this alongside the other publish calls, makes me think if we're specific with those (*_cloud_event_*, *_custom_event*, etc) we should be specific with this too (publish_event_grid_events?)

Reply:
im open to it - but we would need to change the swagger